core/src/Streamly/Internal/Data/Fold/Type.hs

```diff
@@ -829,6 +863,9 @@ splitWith func (Fold stepL initialL extractL) (Fold stepR initialR extractR) =
                 Partial sR -> extractR sR
                 Done rR -> return rR
 
+    final (SeqFoldR f sR) = finalR sR >>= pure . SeqFoldR f
+    final (SeqFoldL sL) = finalL sL >>= pure . SeqFoldL
```

Comment:
If the finalize type was `s -> m b`. The implementation would've been semantically different here.

In `extract` we do `initialR`. Shouldn't we also finalize it there?